Add stand-up extension

theomjones opened the pull request:
## Description

<!-- A summary of your change. If you add a new extension or command, explain what it does. -->

## Screencast

<!-- If you add a new extension or command, include a screencast (or screenshot for straightforward changes). A good screencast will make the review much faster - especially if your extension requires registration in other services.  -->

## Checklist

- [x] I read the [extension guidelines](https://developers.raycast.com/basics/prepare-an-extension-for-store)
- [x] I read the [documentation about publishing](https://developers.raycast.com/basics/publish-an-extension)
- [x] I ran `npm run build` and [tested this distribution build in Raycast](https://developers.raycast.com/basics/prepare-an-extension-for-store#metadata-and-configuration)
- [x] I checked that files in the `assets` folder are used by the extension itself
- [x] I checked that assets used by the `README` are placed outside of the `metadata` folder

pernielsentikaer reviewed (changes requested):
Hi 👋 

Thanks for your contribution 💪 

I have now tested your extension, and I have some feedback ready for you:

- What about having a title field as textfield and a Note field as textarea so, there is more space available?

- Maybe you can add [Draft](https://developers.raycast.com/api-reference/user-interface/form#drafts) support, but it might only work for `Take Note`

- Add support for creating new labels

- Add so it’s possible to edit a note

- Add [confirmAlert](https://developers.raycast.com/api-reference/feedback/alert#confirmalert) before deleting all items

- use [useSql](https://developers.raycast.com/utilities/react-hooks/usesql) instead of `lowdb`, this way we can also tweak useSql if something pops up (should be possible)

I'm looking forward to testing this extension again 🔥 

Request a new review when you are ready. Feel free to contact me here or at [Slack](https://www.raycast.com/community) if you have any questions.

theomjones commented:
> Hi 👋
> 
> Thanks for your contribution 💪
> 
> I have now tested your extension, and I have some feedback ready for you:
> 
> * What about having a title field as textfield and a Note field as textarea so, there is more space available?
> * Maybe you can add [Draft](https://developers.raycast.com/api-reference/user-interface/form#drafts) support, but it might only work for `Take Note`
> * Add support for creating new labels
> * Add so it’s possible to edit a note
> * Add [confirmAlert](https://developers.raycast.com/api-reference/feedback/alert#confirmalert) before deleting all items
> * use [useSql](https://developers.raycast.com/utilities/react-hooks/usesql) instead of `lowdb`, this way we can also tweak useSql if something pops up (should be possible)
> 
> I'm looking forward to testing this extension again 🔥
> 
> Request a new review when you are ready. Feel free to contact me here or at [Slack](https://www.raycast.com/community) if you have any questions.

👋 Hi, thanks for feeding back 

I've address some of your comments and have some comments for others:

- [x] Added description field for more information
- [ ] Draft support, I'm not sure this needs drafts as it's supposed to be very quick, like just a thought during the day that needs logging.
- [ ] I'll iterate after first release for adding labels unless you think it's essential for first release 
- [x] Added edit functionality
- [x] Added confirm for deleting all notes in a day
- [ ] I'm not sure the overhead of adding sql and the implications it brings (migrations, potential ORM inclusion) is worth it

I'm happy to address the unchecked points if you think it's worth it 😃 

Thanks very much

pernielsentikaer commented on extensions/stand-up/src/components/note-form.tsx:
```diff
@@ -0,0 +1,63 @@
+      />
+      <Form.TextArea defaultValue={initialData.description} id="description" title="Description" />
+      <Form.DatePicker id="date" title="Date" defaultValue={initialData.date} />
+      <Form.Dropdown title="Note type" id={"type"} defaultValue={initialData.type}>
```
```suggestion
      <Form.Dropdown title="Note Type" id={"type"} defaultValue={initialData.type}>
```

pernielsentikaer reviewed (changes requested):
Just checked it again and you might want to use something like [useCachedPromise](https://developers.raycast.com/utilities/react-hooks/usecachedpromise) instead of `useAsync`, there you get `cache` and `loading` property directly - the current way also produces [empty state flicker](https://developers.raycast.com/basics/prepare-an-extension-for-store#empty-states) when loading notes

I'm looking forward to testing this extension again 🔥 

Request a new review when you are ready. Feel free to contact me here or at [Slack](https://www.raycast.com/community) if you have any questions.

theomjones commented:
Hey, I've amended this to use the cached promise hook and removed the react-use dependency!

Thanks for your feedback 😃

pernielsentikaer commented:
Checking this out now 🙂

pernielsentikaer commented on extensions/stand-up/src/components/edit-note.tsx:
```diff
@@ -0,0 +1,38 @@
+        key: "e",
+        modifiers: ["cmd"],
+      }}
+      title={"Edit Note"}
```
Missing an icon

pernielsentikaer commented on extensions/stand-up/src/show-notes.tsx:
```diff
@@ -0,0 +1,229 @@
+      navigationTitle="Your standup notes"
+      searchBarAccessory={
+        <List.Dropdown onChange={setFilter} tooltip="Filter by day">
+          <List.Dropdown.Item title="Show all" value="all" />
```
```suggestion
          <List.Dropdown.Item title="Show All" value="all" />
```

pernielsentikaer commented on extensions/stand-up/src/show-notes.tsx:
```diff
@@ -0,0 +1,229 @@
+      isLoading={isLoading}
+      navigationTitle="Your standup notes"
+      searchBarAccessory={
+        <List.Dropdown onChange={setFilter} tooltip="Filter by day">
```
```suggestion
        <List.Dropdown onChange={setFilter} tooltip="Filter by Day">
```

pernielsentikaer reviewed (changes requested):
I added a note today and then I changed it for tomorrow, it still says today here

![Raycast-sQPL2mc6@2x](https://github.com/raycast/extensions/assets/38033107/db8b5ea1-1b1d-497a-91ed-559c4989f91d)

I'm looking forward to testing this extension again 🔥
